pkgs/applications/virtualization/virtualbox/guest-additions/builder.nix

```diff
@@ -21,17 +19,21 @@ in stdenv.mkDerivation (finalAttrs: {
 
   env.NIX_CFLAGS_COMPILE = "-Wno-error=incompatible-pointer-types -Wno-error=implicit-function-declaration";
 
-  nativeBuildInputs = [ patchelf makeWrapper pkg-config which yasm ];
-  buildInputs =  kernel.moduleBuildDependencies ++ [ libxslt libX11 libXext libXcursor
-    glib nasm alsa-lib makeself pam libXmu libXrandr linuxHeaders openssl libpulseaudio xorg.xorgserver ];
+  nativeBuildInputs = [ patchelf pkg-config which yasm makeself nasm xorg.xorgserver openssl linuxHeaders xz ] ++ kernel.moduleBuildDependencies;
```

Comment:
Does it really need yasm _and_ nasm? Usually one of them is fine.

Reply:
Shouldn't `xz`, `openssl`, `linuxHeaders` be in `buildInputs`?

Reply:
Also `xorg.xorgserver` should probably be in `buildInputs` as it was before.

Reply:
xz is not shown in ldd, yet a build dependency
openssl is not shown in ldd, yet a build dependency
linuxHeaders should only be required when prebuilding the guest additions
xorg.xorgserver should only be required while building

Reply:
> Does it really need yasm _and_ nasm? Usually one of them is fine.

nasm is now removed